### PR TITLE
fix: SIP-15 time_range_endpoints not using default option

### PR DESCRIPTION
**Pull request by rubenSastre**
### SUMMARY
since SIP-15 is enabled, the time_range_endpoints is hardcoded to [start,end), without possibility of taking value set on extra params on database, or on default parameter on conf.py

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

image

image

### ADDITIONAL INFORMATION

- [x ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rubenSastre** [comment]
@villebro could you have a look?

**Comment by villebro** [comment]
Thanks @rubenSastre , I'll take a look as soon as I can! (marking myself as reviewer to not forget)

File: superset/config.py [hunk 1]
```diff
@@ -950,7 +950,7 @@ class CeleryConfig:  # pylint: disable=too-few-public-methods
 # indefinite.
 SIP_15_ENABLED = True
 SIP_15_GRACE_PERIOD_END: Optional[date] = None  # exclusive
-SIP_15_DEFAULT_TIME_RANGE_ENDPOINTS = ["unknown", "inclusive"]
+SIP_15_DEFAULT_TIME_RANGE_ENDPOINTS = ["inclusive", "exclusive"]
```
Comment (villebro) [inline superset/config.py:953]: I'm surprised these are not in line with the comment in `utils`. I would be interested in hearing @john-bodley's thoughts on this change, as it looks like this is a bug.

**Review by villebro — APPROVED** [review]
Thanks for the PR @rubenSastre and sorry for the long review time. I agree with the changes made here, but would still like to hear @john-bodley chime in before merging.

**Comment by john-bodley** [comment]
@rubenSastre I believe there's a disconnect between you PR description and code change. The existing code (at least in the places you changed) seems correct to me, i.e., we always want to ensure new slices and all slices after the grace period ends adhere to the `[start, end)` interval definition. 

More specifically, the target (desired) time interval is purposely not configurable (and thus is hard coded). The source intervals may vary by institution and can be defined at the database level or fallback to the configuration file if undefined per [here](https://github.com/apache/incubator-superset/blob/4d329071a1651994574c3ad1f26e32b56d9ff810/superset/views/utils.py#L285-L301).

What is the specific issue you're facing? The disconnect in your two screenshots could be because either:

1. The slice is a new slice.
2. The grace period has expired.

**Comment by rubenSastre** [comment]
@john-bodley , what I detect is that the database extra value and/or the DEFAULT_TIME_RANGE_ENDPOINT is not working, all the new and the existing charts will be force to be [start, end).  This should be the default rule, but we should be able to give the option of setting a different rule for the endpoints.   

My changes will use the DEFAULT_TIME_RANGE_ENDPOINT as the value to use for the new charts and to popup the alert of updating if having a different while the the grace period is set.

**Comment by john-bodley** [comment]
@rubenSastre per your comment,

> ... , all the new and the existing charts will be force to be [start, end). 

This is the intended behavior, i.e., the time range endpoints are not configurable and per SIP-15 ensure that `[start, end)` is used consistency throughout Superset for all connectors (SQL and the Druid REST API) as previously this wasn't the case. You can refer to [SIP-15](https://github.com/apache/incubator-superset/issues/6360) regarding the motivation for this change.

**Comment by rubenSastre** [comment]
@john-bodley ,reading this, so if we don´t want to be an optional configuration has not sense, as what I did is to use the SIP_15_DEFAULT_TIME_RANGE_ENDPOINTS to give this option.

By the way then, the extra parameter for database option doesn´t work when a new chart is created, could you have a look? I don´t know which is the best option as some times the chart_id doesn´t exist, at the begining so [start, end) is set on the creation.

**Comment by john-bodley** [comment]
@rubenSastre the `SIP_15_DEFAULT_TIME_RANGE_ENDPOINTS` and database overrides exist merely so deployments can define how time range intervals were defined for said environment pre-SIP-15, i.e., if a database was using timestamps the time range may have been behaving as `[inclusive, inclusive]` whereas if the database was using date strings the time range may have been behaving as `[exclusive, inclusive]`. With SIP-15 enabled all time ranges (regardless of connector or database) behave as `[inclusive, exclusive)` and are not configurable.

**Comment by yongchand** [comment]
@john-bodley I feel like a lot of people want [inclusive, inclusive] according to slack community, including me. Are you still concrete with opinion for [inclusive, exclusive)? What is the point of `SIP_15_DEFAULT_TIME_RANGE_ENDPOINTS` exactly?

**Pull request closed without merging**